### PR TITLE
Added algolia with public api keys

### LINKED ISSUE #24: Incorporate a search function into the docs
In order to achieve parity with the legacy docs, we should incorporate a search function. The search functionality should be similar in behavior to what is currently possible, including:

* Keyword search
* [Autocomplete](https://www.algolia.com/blog/ux/what-are-predictive-search-and-autocomplete/) search results (results displayed as you type)

Screenshot 2023-02-06 at 3 35 09 PM

### PR DESCRIPTION
**Pull request by shanejonas**
fixes #24

### CODE REVIEW

File: package-lock.json
Comment (rekmarks) [inline package-lock.json:1]: Should this file be deleted?

**Comment by vandan** [comment]
Looks good to me. The new hint for the keyboard shortcut in the search bar is pretty helpful.

**Review by adonesky1 — APPROVED** [review]
Looks great!

**Review by BelfordZ — APPROVED** [review]

**Review by BelfordZ — APPROVED** [review]

**Review by BelfordZ — APPROVED** [review]

**Comment by BelfordZ** [comment]
double approved, boom

**Pull request closed without merging**